x-pack/plugin/esql/build.gradle

```diff
@@ -21,6 +21,7 @@ dependencies {
   implementation project('compute:ann')
   implementation project(':libs:elasticsearch-dissect')
   implementation project(':libs:elasticsearch-grok')
+  // Also contains a dummy processor to allow compilation with unused annotations.
```

Comment:
was this intentional?

Reply:
Yes.

The `ConsumeProcessor` is a necessary hack to make compilation pass as the `@ServerlessScope` annotation is unused in this repo itself. Otherwise we'd need to add compiler options `-Xlint:-processing`.

I had no idea that's the case and only learned this from Nik - so I think it's useful to leave a comment for the future.